cherrypy/process/wspbus.py

```diff
@@ -342,16 +342,10 @@ def block(self, interval=0.1):
             # that another thread executes cherrypy.engine.exit()
             if (
                     t != threading.currentThread() and
-                    t.isAlive() and
                     not isinstance(t, threading._MainThread)
             ):
                 # Note that any dummy (external) threads are always daemonic.
-                if hasattr(threading.Thread, 'daemon'):
-                    # Python 2.6+
-                    d = t.daemon
-                else:
-                    d = t.isDaemon()
-                if not d:
+                if not t.daemon:
```

Comment:
Please move this condition to the parent `if`-block as nesting isn't needed anymore.